app/Actions/CreateAndAttachRating.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Actions;
+
+use App\Models\Entry;
+use App\Models\Rating;
+use Illuminate\Support\Facades\Auth;
+use Lorisleiva\Actions\Concerns\AsAction;
+
+class CreateAndAttachRating
+{
+    use AsAction;
+
+    public function handle(Entry $entry, array $ratingData)
+    {
+        $rating = Rating::create([
+            'entry_id' => $entry->id,
+            'user_id' => $entry->dungeon_master_id,
+            'author_id' => $entry->user_id,
+            'categories' => 0,
+        ]);
+        foreach ($ratingData as $category => $value) {
+            if ($category == 'creative' && $value) {
+                $rating->addCategory(CREATIVE_BITMASK);
+            } elseif ($category == 'flexible' && $value) {
+                $rating->addCategory(FLEXIBLE_BITMASK);
+            } elseif ($category == 'friendly' && $value) {
+                $rating->addCategory(FRIENDLY_BITMASK);
+            } elseif ($category == 'helpful' && $value) {
+                $rating->addCategory(HELPFUL_BITMASK);
+            } elseif ($category == 'prepared' && $value) {
+                $rating->addCategory(PREPARED_BITMASK);
+            }
+        }
```

Comment:
we can probably simplify this a ton by using a map (like [in this refactor](https://github.com/DnD-Montreal/session-tome/pull/220/commits/8c02a84d611540f772fb47878877c4cc9f601d14)) and string interpolation like this: 
```php
if ($value) {
    $key = strtoupper($category) . "_LABEL";
    $rating->addCategory($ratingMap[$key])
}
````

Reply:
Also you're not referencing the class these constants come from... so you'd need to do: 
```suggestion
        foreach ($ratingData as $category => $value) {
            if ($category == 'creative' && $value) {
                $rating->addCategory(Rating::CREATIVE_BITMASK);
            } elseif ($category == 'flexible' && $value) {
                $rating->addCategory(Rating::FLEXIBLE_BITMASK);
            } elseif ($category == 'friendly' && $value) {
                $rating->addCategory(Rating::FRIENDLY_BITMASK);
            } elseif ($category == 'helpful' && $value) {
                $rating->addCategory(Rating::HELPFUL_BITMASK);
            } elseif ($category == 'prepared' && $value) {
                $rating->addCategory(Rating::PREPARED_BITMASK);
            }
        }
```

Reply:
Both suggestions added.